improver/metadata/check_datatypes.py

```diff
@@ -38,46 +38,46 @@
     TIME_COORD_NAMES, TIME_INTERVAL_DTYPE, TIME_INTERVAL_UNIT,
     TIME_REFERENCE_DTYPE, TIME_REFERENCE_UNIT)
 
+DESIRED_FLOAT_DTYPE = np.float32
 
-def check_cube_not_float64(cube, fix=False):
-    """Check a cube does not contain any float64 data, excepting time
-    coordinates. The cube can be modified in place, if the fix keyword is
-    specified to be True.
+
+def ensure_cube_floats_are_float32(cube, fix=False):
+    """Ensure that any floating point data are of dtype float32.
+    The cube can be modified in place, if the fix keyword is specified to be
+    True.
 
     Args:
         cube (iris.cube.Cube):
-            The input cube that will be checked for float64 inclusion.
+            The input cube that will be checked for float32 non-conformance of
+            floating point data.
         fix (bool):
-            If fix is True, then the cube is amended to not include float64
-            data, otherwise, an error will be raised if float64 data is found.
+            If fix is True, the cube is amended to cast floating point data to
+            float32, otherwise, an error will be raised if non-float32 floating
+            point data is found.
 
     Raises:
-        TypeError : Raised if 64 bit values are found in the cube.
+        TypeError : Raised if non 32-bit dtypes are found in the cube.
     """
-    if cube.dtype == np.float64:
-        if fix:
-            cube.data = cube.data.astype(np.float32)
-        else:
-            raise TypeError("64 bit cube not allowed: {!r}".format(cube))
 
-    for coord in cube.coords():
-        if coord.units.is_time_reference():
-            continue
+    def dtype_invalid(obj):
+        return (np.issubdtype(obj.dtype, np.floating) and
+                obj.dtype != DESIRED_FLOAT_DTYPE)
```

Comment:
This looks better, just check the float types, time may not be the only need for high precision ints